### DISCUSSION TITLE
Difficulty Automatically Fetching Firestore Database Structure

### DISCUSSION THREAD

**Discussion by niko2020** [Budibase/budibase#14833] [Help] [answered]
Hello everyone,

I'm currently testing Budibase to create a back office connected to Firestore. However, I noticed that it doesn't seem to automatically fetch the Firestore database structure, which makes the setup quite challenging for non-database experts like myself.

I'm wondering if there’s an option or a method to automate the import of the collections and documents structure, or if there are any tools/plugins available to simplify this process. Any help or clarification would be greatly appreciated!

Thanks in advance.

**Comment by mikesealey** [comment, marked as answer]
Hey @niko2020 

Budibase accesses data from non-relational data sources like Firestore using queries. You'll need to connect to your Firestore, and then write a query to display the information you want. 

There are a few examples in our [documentation here](https://docs.budibase.com/docs/firestore) which will help get you started.

**Reply by niko2020** [reply]
Thank you very much, but this process feels quite cumbersome (especially compared to what platforms like Jetadmin offer).

I'm also encountering significant issues as some collections seem to be inaccessible, even though I’ve configured open rules in Firebase.

Here’s the error I’m receiving:

javascript
Copier le code
TypeError: function () {
    const temp = Object.assign({}, this);
    if (temp.credentials) {
    ...<omitted>...
} could not be cloned.
